infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -56,27 +80,119 @@ def run_shell_command(command, env_vars):
 ###################
 
 
-def make_backup_fpath():
-    return BACKUP_DIR / datetime.now().strftime(FILENAME_DATETIME_FORMAT)
+def make_backup_object_name_from_datetime() -> str:
+    """Creates a string using the format specified by the FILENAME_DATETIME_FORMAT
+    global variable for using as the filename of the backups.
+
+    :return: the string to be used for the backup file name
+    """
+    return datetime.now().strftime(FILENAME_DATETIME_FORMAT)
+
+
+def make_backup_fpath(object_name: str) -> Path:
+    """Prepends the backup path to give a filepath for the object.
+
+    :param object_name: the name of an object to which the BACKUP_DIR
+        global variable should be prepended, defaults to None
+
+    :return: returns a :class:'Path' object that gives a filepath
+        for the 'object_name' file in the BACKUP_DIR directory
+    """
+    return BACKUP_DIR / object_name
+
+
+def create_s3_session() -> boto3.session.Session:
+    """Creates and returns an AWS session for listing and downloading backups in S3.
+
+    :return: returns a :class:'boto3.session.Session' object using the AWS_ACCESS_KEY_ID
+    and AWS_SECRET_ACCESS_KEY global variables that are set from the environment
+    variables of the same name for interacting with AWS
+    """
+    sess = boto3.session.Session(
+        aws_access_key_id=AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+    )
+    return sess
+
+
+def create_s3_client() -> BaseClient:
+    """Creates and returns an AWS S3 client for uploading backup to the cloud.
+
+    :return: returns a :class:'BaseClient' object for uploading the backup
+        file to AWS S3
+    """
+    sess = create_s3_session()
+    return sess.client("s3")
+
+
+def upload_backup_to_s3(
+    s3_client: BaseClient,
+    backup_fpath: Path,
+    backup_bucket_name: str,
+    backup_object_name: str,
+):
+    """Uploads the backup_fpath file to AWS S3.
+
+    :param s3_client: an S3 client to use to upload the file, defaults to None
+    :param backup_fpath: the filepath to the file that will be uploaded
+        , defaults to None
+    :param backup_bucket_name: the S3 bucket to upload the file to
+        , defaults to None
+    :param backup_object_name: the filename for the uploaded object in S3
+        , defaults to None
+    """
+    with open(str(backup_fpath), "rb") as f:
+        s3_client.upload_fileobj(Fileobj=f, Bucket=backup_bucket_name, Key=backup_object_name)
+
+
+def delete_local_backup_file(backup_fpath: Path):
+    """Deletes the local backup file once it has been uplaoded to S3.
+
+    :param backup_fpath: the filepath to the file that will be deleted
+        , defaults to None
+    """
+    os.remove(str(backup_fpath))
+
 
+def backup_database(backup_object_name: str):
+    """Creates a local backup of the database, uploades it to S3, and delets the local backup.
 
-def backup_database(db_backup_gzip_fpath):
+    :param backup_object_name: the name to be used for the backup file
```

Comment:
Is it true that this `backup_object_name` defaults to `None`? If so, the signature should be `backup_object_name: Optional[str] = None`. If that isn't true, you can just remove the `defaults to None` statement.